src/datasets/commands/test.py

```diff
@@ -93,7 +93,7 @@ def run(self):
             print("Both parameters `config` and `all_configs` can't be used at once.")
             exit(1)
         path, name = self._dataset, self._name
-        module_path, hash = prepare_module(path)
+        module_path, hash = prepare_module(path, cache_dir=self._cache_dir)
```

Comment:
The datasets library uses three caches: one to store the datasets data, one to store the metrics data and one to store the modules (either datasets or metrics python scripts).

By default each one of these caches is a directory in the huggingface cache home directory (by default `~/.cache/huggingface`). The three directories are named `datasets`, `metrics` and `modules`. We try to have them separate so that if users ever want to clear the data cache but keeping the modules for example they can do that easily.

In the `test.py` script the `--cache_dir` refers to the `datasets` cache dir since it is passed as `cache_dir` to the dataset builders.
However here `prepare_module` expects the `modules` cache dir.


Maybe in `test.py` we can change `--cache_dir` to be the huggingface cache home directory and then pass `cache_dir / "datasets"` to the builders and `cache_dir / "modules"` to `prepare_modules` ?

Reply:
One more thing: `prepare_module` also expects the cache dir to be the one for downloaded files (default is `~/.cache/huggingface/datasets/downloads`).

Maybe to make things clearer we should keep `cache_dir` as the cache used to download files and store datasets data files, and instead add a `dynamic_module_path` parameter that defaults to `DYNAMIC_MODULES_PATH`.
I'd like to keep them separate in terms or variable name to avoiding making things confusing.